app/services/db/monthly_sp_auth_count/total_monthly_auth_counts_within_iaa_window.rb

```diff
@@ -26,10 +26,13 @@ def call(service_provider)
         # - full months from monthly_sp_auth_counts
         # - partial months by aggregating sp_return_logs
         # The results are rows with [ial, auth_count, year_month, issuer, iaa, iaa start, iaa end]
-        union_query = [
-          full_month_subquery(sp: service_provider, full_months: full_months),
-          *partial_month_subqueries(sp: service_provider, partial_months: partial_months),
-        ].join(' UNION ALL ')
```

Comment:
good catch! what if we did `.compact.join(' UNION ALL ')` or `.select(&:present?).join(' UNION ALL ')`

Reply:
Or honestly even just `*full_month_subquery` would have the same effect

Reply:
The full_month_subquery returns a string not an array and the string is not empty when there are no full_months.

Reply:
what if we updated that method to return nil when there are no `full_months`?